base/pkg/types.jl

```diff
@@ -40,15 +41,73 @@ show(io::IO, s::VersionSet) = join(io, s.intervals, " ∪ ")
 isempty(s::VersionSet) = all(isempty, s.intervals)
 in(v::VersionNumber, s::VersionSet) = any(i->in(v,i), s.intervals)
 function intersect(A::VersionSet, B::VersionSet)
-    ivals = vec([ intersect(a,b) for a in A.intervals, b in B.intervals ])
-    ivals = copy(ivals) # temporary bandaid for issue #4592
+    ivals = [intersect(a,b) for a in A.intervals for b in B.intervals]
```

Comment:
It's already fairly granular. Also note that I had to adjust an assert to avoid anonymous type parameter syntax, I think https://github.com/JuliaLang/julia/commit/f907b5716805013fcd65b1720d1ff940e11ed815#diff-cac60f612c90efb7c227b01575fb8e15R214 should be equivalent